### ISSUE TITLE
Bionic not applying to ticked cardtypes, "Toggle All Notetypes" doesn't seem to work either.

### ISSUE THREAD

**Issue by natnina** [AnKing-VIP/anki-bionic-reading#6]
#### Problem description

*Please describe the issue concisely in here. In case of an error: Walk us through the steps you took to get there. What happened? What did you expect to happen?*

Problem 1: The "Toggle All Notetypes" doesn't seem to do anything/doesn't toggle all notetypes. 
Problem 2: The Bionic reading doesn't seem to be applying to all the cards that I ticked with a checkmark. I tried just with "Cloze" cards but it didn't apply. I've restarted Anki a few times after downloading, after receiving the error (pasted below), and after I reclicked note types "on". 
Problem 3: After restarting, the ticked notetypes "unclicks". 

#### Checklist

*Please replace the space inside the brackets with an **x** if the following items apply:*

 - [x] I've restarted Anki to see if it helps
 - [x] I've verified that I use the latest version of the add-on by redownloading it from AnkiWeb
 - [x] I've tried to disable other add-ons to see if there are any interactions present
 - [ ] My issue disappears when I hold shift while starting Anki.
 - [x] I've checked if anyone else reported this problem before by looking through the issue reports. I also checked to see if there is a section about known issues in the add-on description, documentation, or README.
 

#### What Anki version are you using?

*Please open Anki, go to Help → About (windows) or Anki → About Anki (mac) to find the version. Edit the value between the backticks below

```
2.1.65
```

#### Error message (if any)

*If you've received an error message, please copy and paste it between the backticks below:*

```
Error
An error occurred. Please start Anki while holding down the shift key, which will temporarily disable the add-ons you have installed.
If the issue only occurs when add-ons are enabled, please use the Tools > Add-ons menu item to disable some add-ons and restart Anki, repeating until you discover the add-on that is causing the problem.
When you've discovered the add-on that is causing the problem, please report the issue to the add-on author.
Debug info:
Anki 2.1.65 (aa9a734f) Python 3.9.15 Qt 6.5.0 PyQt 6.5.0
Platform: macOS-12.7-x86_64-i386-64bit
Flags: frz=True ao=True sv=3
Add-ons, last update check: 2023-11-12 11:41:49
Add-ons possibly involved: ⁨Bionic Reading⁩

Traceback (most recent call last):
  File "aqt.main", line 607, in loadCollection
  File "_aqt.hooks", line 1497, in __call__
  File "/Users/natashac.k./Library/Application Support/Anki2/addons21/649714433/main.py", line 39, in on_anki_start
    check_update()
  File "/Users/natashac.k./Library/Application Support/Anki2/addons21/649714433/firstrun.py", line 77, in check_update
    on_update(prev_version, curr_version)
  File "/Users/natashac.k./Library/Application Support/Anki2/addons21/649714433/firstrun.py", line 68, in on_update
    add_script_to_all_note_types()
  File "/Users/natashac.k./Library/Application Support/Anki2/addons21/649714433/notetype.py", line 54, in add_script_to_all_note_types
    mw.col.models.update_dict(note_type)
  File "anki.models", line 223, in update_dict
  File "anki._backend_generated", line 862, in update_notetype_legacy
  File "anki._backend", line 156, in _run_command
anki.errors.CardTypeError: Card template ⁨3⁩ in notetype '⁨A. Artery⁩' has a problem.<br>Expected to find a field replacement on the front of the card template.
```

**Comment by FancyGhosty** [comment]
I am having the same issue.

**Comment by BlueGreenMagick** [comment]
I think your note type may already be invalid. Can you try opening the note type template in Anki Note types and see if it saves alright?

**Comment by AnKingMed** [comment]
@BlueGreenMagick are you ok to troubleshoot this? I'm not quite sure what you thought was going on

**Comment by BlueGreenMagick** [comment]
1. In toolbar, go to ' Tools > Manage Note Types '. 
2. Select 'A. Artery' notetype and click 'Cards...' button.
3. In the card type editor, change 'Card Type' to the 3rd template. 
4. In Front template, you need to add a field replacement. (e.g. `{{Front}}`)

**Comment by natnina** [comment]
![Screen Shot 2024-01-01 at 6 25 26 PM](https://github.com/AnKing-VIP/anki-bionic-reading/assets/130001038/27850bff-6626-4ff7-9f74-425cef2badd0)

New problem in the preview window "couldn't find Front" so I changed to {{Text}} and same message.

**Comment by BlueGreenMagick** [comment]
That's because your note type doesn't have the field 'Front'.

I think you accidentally added a new Card template and what you want to do is to delete it. Click the 'Options...' button on the right side of the 'Card Type' bar and click 'Remove Card Type...'

**Comment by natnina** [comment]
> That's because your note type doesn't have the field 'Front'.
> 
> I think you accidentally added a new Card template and what you want to do is to delete it. Click the 'Options...' button on the right side of the 'Card Type' bar and click 'Remove Card Type...'

I didn't add any new Card Templates I only use what is included in the deck download. I don't feel comfortable deleting this card template since there are 269 notes that use it (screenshot below).

Following your instructions, I clicked the third (I think template) option in Card Type (you can see that on the previous screenshot)... so if not {{front}}, what would I add? 

> @BlueGreenMagick are you ok to troubleshoot this? I'm not quite sure what you thought was going on

I'm not sure what you thought was going on either. Would you elaborate on the full scope, please?

![image](https://github.com/AnKing-VIP/anki-bionic-reading/assets/130001038/7b518e29-4ab0-4f18-b232-e4475a2319e3)

**Comment by AnKingMed** [comment]
None of your fields in that picture are "Front". You have "Artery" "Source" etc. THat's probably why. You need to add one of those field names and not {{Front}}

**Comment by natnina** [comment]
Got it - it worked for that A.Artery note type. I copied and pasted what was on Card Type template 1 to the dropdown template 3. 
I ticked Anking Overhaul (Anking/AnkingMed) notetypes and it works too. Thank you!
However, other notetypes do not remain ticked after reopening Anki. Nor shows bionic reading even though it shows the field.

example: 
These were manually ticked before closing (the Toggle All Notetype button still doesn't work):
![Screen Shot 2024-01-03 at 4 48 46 PM](https://github.com/AnKing-VIP/anki-bionic-reading/assets/130001038/58992131-c77a-4491-b9e8-63e72f5a38fd)

This is the only one that remained when reopening anki:
![Screen Shot 2024-01-03 at 5 13 16 PM](https://github.com/AnKing-VIP/anki-bionic-reading/assets/130001038/8e092218-d133-4552-a8bf-02fef999e8cd)

I checked the notetype of AnkiDerm specifically. It seems to have the cloze deletion for the field already on there. How should I proceed?? I'm super new to this so please excuse my limited knowledge on this.
![image](https://github.com/AnKing-VIP/anki-bionic-reading/assets/130001038/34aad6ac-0115-4289-b505-8f4e37b9429b)

**Comment by BlueGreenMagick** [comment]
Did you click save after changing add-on config?

**Comment by natnina** [comment]
Yup!

**Comment by BlueGreenMagick** [comment]
Hmm... might be because those are AnkiHub notetypes. AnkiHub add-on may be syncing the notetypes on startup so the edits to notetypes by bionic reading are being reverted. @AnKingMed may know more.

**Comment by AnKingMed** [comment]
@BlueGreenMagick that doesn't make sense because even the AnKingOverhaul note type that isn't an AnkiHub note type is getting unchecked and the AnkiHub one is staying? (but only one of the AnkiHub note types..)

@natnina I've noticed 2.1.65-66 has lots of weird bugs. Can you try 2.1.64 or 23.10?  Also, does this still occur if Bionic Reading is the only addon toggled to be enabled and all others are disabled?

**Comment by natnina** [comment]
@AnKingMed  Since my first post about this in November, I've updated to 23.10, and the issue persists. I think I've been on 23.10 for a few weeks now.
![image](https://github.com/AnKing-VIP/anki-bionic-reading/assets/130001038/c98acbbb-e40d-42cf-8551-4d216940bd70)

Yes. I disabled all my add-ons except for Bionic Reading. Restarted. Opened BIonic Reading and ticked the 3 Anking notetypes. I noticed it's after I press "Sync" that all the toggles are being removed (including "AnkingOverhaul (Anking/Ankingmed)"!) So after each Sync and Restart, I have to reopen Bionic Reading and manually tick again.
